types/smb2/index.d.ts

```diff
@@ -0,0 +1,54 @@
+// Type definitions for smb2 0.2
+// Project: https://github.com/bchelli/node-smb2
+// Definitions by: Nick Whitlock <https://github.com/eezstreet>
+// Definitions: https://github.com/eezstreet/DefinitelyTyped
+
+export = SMB2;
+
+interface SMB2Options {
+    share: string;
+
+    domain: string;
+
+    username: string;
+
+    password: string;
+
+    port?: number;
+
+    packetConcurrency?: number;
+
+    autoCloseTimeout?: number;
+}
+
+interface SMB2ReadFileOptions {
+    encoding: string | null;
+}
+
+declare class SMB2 {
+    constructor(options: SMB2Options);
```

Comment:
```suggestion
    constructor(options?: SMB2Options);
```
https://github.com/bchelli/node-smb2/blob/master/lib/smb2.js#L24